src/main/scala/com/banzaicloud/spark/metrics/sink/PrometheusSink.scala

```diff
@@ -279,9 +297,31 @@ abstract class PrometheusSink(property: Properties,
     }
   }
 
+  /**
+    * 2 sources for labels:
+    *  1. from SparkConf
+    *  2. property labels
+    * @return
+    */
+  private def collectLabels(sparkConf: SparkConf): Map[String, String] = {
```

Comment:
```suggestion
  private def collectLabels(sparkConf: SparkConf): Try[Map[String, String]] = {
```
As parseLabels, may throw an exception.